File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int mc = nums[0];
        int count = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == mc) {
                count++;
            } else {
                count--;
                if (count == 0) {
                    mc = nums[i];
                    count = 1;
                }
            }
        }

        return mc;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mc = 0
        count = 0

        for num in nums:
            if count == 0:
                mc = num
            
            if num == mc:
                count += 1
            else:
                count -= 1

        return mc